### PR TITLE
Add hook to clean up YAML files

### LINKED ISSUE #410: Lint YAML files
It seems that YAML files are not being linted and auto-fixed by `pre-commit`. This functionality should be added.

- `pre-commit` should lint and auto-format YAML files
- Lint check should also check PRs for the same rule

For example, consider this one-off 4-space indentation in a file that is otherwise indented by 2 spaces.
https://github.com/WordPress/openverse-api/blob/e2c490ad9d0f6ca47e5e2f4fd06205b7eac13c45/docker-compose.yml#L60
Pre commit should have fixed it and, if not, the linting check should have caught it.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #410 by @dhruvkb

## Description

This PR adds a new `pre-commit` hook to format YAML files.

## Testing Instructions

Formatting changes are adequately tested by passing lint check.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by AetherUnbound — APPROVED** [review]
Looks good! Do we want to add this to the main Openverse repo?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,51 +1,56 @@
...
+  rev: v4.0.1
+  hooks:
+  - id: trailing-whitespace
+  - id: check-executables-have-shebangs
+  - id: check-json
+  - id: check-case-conflict
+  - id: check-toml
+  - id: check-merge-conflict
+  - id: check-xml
+  - id: check-yaml
```
Comment (sarayourfriend) [inline .pre-commit-config.yaml:14]: We can remove this check now I think 🙂

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,51 +1,56 @@
...
+  rev: 21.6b0
+  hooks:
+  - id: black
+    args:
+    - --safe
+
+- repo: https://github.com/dhruvkb/ymlfmt
+  rev: 66026fc
+  hooks:
+  - id: ymlfmt
```
Comment (sarayourfriend) [inline .pre-commit-config.yaml:56]: An alternative to having to maintain a separate repository for yaml formatting (including updates, etc) we could use the [prettier pre-commit hook](https://prettier.io/docs/en/precommit.html#option-3-pre-commithttpsgithubcompre-commitpre-commit) which already supports yaml.

```suggestion
- repo: https://github.com/pre-commit/mirrors-prettier
  rev: v2.5.1
  hooks:
  - id: prettier
    types_or: [yaml]
```

A bonus is that we already use prettier in WordPress/openverse and WordPress/openverse-frontend.

**Review by obulat — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (dhruvkb) [reply .pre-commit-config.yaml:56]: This is great, I wish I knew sooner. I'll look into this.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (dhruvkb) [reply .pre-commit-config.yaml:14]: `check-yaml` checks for syntactical validity. I suppose that's a good check to have?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (dhruvkb) [reply .pre-commit-config.yaml]: Changed the hook to use Prettier in 58492dc.

**Comment by dhruvkb** [comment]
@AetherUnbound https://github.com/WordPress/openverse/pull/131

**Pull request merged**